### ISSUE TITLE
Wrong message callback destructed

### ISSUE THREAD

**Issue by thedrakes** [SaschaWillems/Vulkan#15]
Isn't this a bug in vulkandebug.cpp:
You got 
	VkDebugReportCallbackEXT msgCallback;
and 
		VkDebugReportCallbackEXT debugReportCallback;
I didn't find documentation on these extensions so I am not sure.

Edit: Maybe you should also consider setting up the debug callback directly after creating the instance instead of after device setup.

**Comment by SaschaWillems** [comment]
Yes, acually nothing gets assigned to msgCallback and this it's never freed. Will check and fix.

**Comment by SaschaWillems** [comment]
Confirmed that this is wrong, but I won't be fixing this yet. My example set the apiversion to 1.0.2, so users with drivers that don't support the most recent api version (e.g. AMD) can use the examples too, but this seems to cause an exception when destroying the debug callback.

I have opened up an [issue](https://vulkan.lunarg.com/app/issues/56c4e9057ef24d0001aae0f5) a the LunarXChange.

So I'll leave this for open now and fix it once the SDK bug is fixed.
